### PR TITLE
[FEATURE] Add fetchCheckout

### PR DESCRIPTION
**Pull request by jzjiang**
This PR adds a `fetchCheckout` method to the client to fetch checkouts by ID.  Most changes are renaming `productQuery` --> `productNodeQuery` and `collectionQuery` --> `collectionNodeQuery` to more accurately reflect the queries.

### CODE REVIEW

File: src/checkout-node-query.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  'taxExempt',
+  'taxesIncluded',
+  'currencyCode',
+  'totalTax',
+  'subtotalPrice',
+  'totalPrice',
+  'completedAt',
+  'createdAt',
+  'updatedAt'
+];
```
Comment (swalkinshaw) [inline src/checkout-node-query.js]: Is this every field on a checkout? Should we include them all by default?

File: src/checkout-node-query.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+export default function checkoutNodeQuery(fields = defaultFields) {
+  return function(parentSelection, fieldName, id) {
+    parentSelection.add(fieldName, {args: {id: createGid('Checkout', id)}}, (node) => {
+      node.addInlineFragmentOn('Checkout', (product) => {
+        addFields(product, fields);
+      });
+    });
+  };
+}
```
Comment (minasmart) [inline src/checkout-node-query.js]: Why is this needed separately from `checkoutQuery`? It looks like there's a lot of shared behaviour.

File: src/client.js [hunk 3]
```diff
@@ -158,7 +160,17 @@ export default class Client {
     });
   }
 
-  fetchCollection(id, query = collectionQuery()) {
+  fetchCollection(id, query = collectionNodeQuery()) {
```
Comment (minasmart) [inline src/client.js]: Should the queries really vary if they're being requested off node, or off of another object? Should we maybe make an abstraction around fragments, and then share common code between fragments and queries for types?

File: src/checkout-node-query.js (hunk 1 above)
Comment (minasmart) [inline src/checkout-node-query.js]: We should probably be striving to request what's required to render a cart and/or a checkout. Is there anything in here that's over and above that?

Comment (swalkinshaw) [inline src/checkout-node-query.js]: Hard to say 🤔  not sure I feel strongly enough about any of them, so let's keep them

Comment (jzjiang) [inline src/checkout-node-query.js]: We probably don't need `ready` if we're not using any of the `checkoutComplete` mutations. I'm not sure if we need `paymentDue` if we're not exposing discount/gift card mutations, `order` or `orderStatusUrl`. Otherwise I think the rest could be used.

Comment (minasmart) [inline src/checkout-node-query.js]: we do need ready. That tells us if taxes have settled out, or if they're still processing. Plus shops need this. We do want to be able to render orders, so I would keep those links.

Comment (minasmart) [inline src/checkout-node-query.js]: maybe just `orderStatusUrl` for now, since that comes with its own UI.

File: src/client.js (hunk 3 above)
Comment (jzjiang) [inline src/client.js]: I'll try to see if I can combine the queries and inline fragments.

**Comment by minasmart** [comment]
looks like this needs a rebase.

File: test/query-test.js [hunk 4]
```diff
@@ -7,7 +7,7 @@ import variantConnectionQuery from '../src/variant-connection-query';
 import optionQuery from '../src/option-query';
 import imageQuery from '../src/image-query';
 import imageConnectionQuery from '../src/image-connection-query';
-import collectionQuery from '../src/collection-query';
+import collectionNodeQuery from '../src/collection-node-query';
```
Comment (minasmart) [inline test/query-test.js]: What does `fetchCollections` use now?

Comment (jzjiang) [inline test/query-test.js]: It uses `collectionConnectionQuery` to fetch multiple collections off shop and `collectionNodeQuery` to fetch a collection off the root.

File: src/collection-node-query.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import imageQuery from './image-query';
+import nodeQuery from './node-query';
+
+const defaultFields = ['id', 'handle', 'updatedAt', 'title', ['image', imageQuery()]];
```
Comment (minasmart) [inline src/collection-node-query.js]: Are these fields duplicated in `collectionConnectionQuery`? If so, you should  probably share them.

Comment (jzjiang) [inline src/collection-node-query.js]: I'll export these fields and update

**Comment by jzjiang** [comment]
Exported default fields in some of the other queries that end up sharing fields (i.e. `image` and `variants`) as well.

**Comment by minasmart** [comment]
👍

**Comment by jzjiang** [comment]
@swalkinshaw do you have any other comments?
